### PR TITLE
[NCL-5671]New 2 step process for product versions in new build-config wizard

### PR DESCRIPTION
**Pull request by DnsZhou**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

**Comment by thescouser89** [comment]
retest this please

**Comment by DnsZhou** [comment]
I have left a description for the detailed design of this component on its jira comments, you can find it [here](https://projects.engineering.redhat.com/browse/NCL-5671?focusedCommentId=2371886&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2371886)

**Comment by DnsZhou** [comment]
retest this please

File: ui/app/build-configs/directives/pnc-create-build-config-product-version-form/pncCreateBuildConfigProductVersionForm.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  angular.module('pnc.build-configs').component('pncCreateBuildConfigProductVersionForm', {
+    templateUrl: 'build-configs/directives/pnc-create-build-config-product-version-form/pnc-create-build-config-product-version-form.html',
+    require: {
+      ngModel: 'ngModel'
+    },
+    controller: [Controller]
+  });
+
+  function Controller() {
+    var $ctrl = this;
```
Comment (matedo1) [inline ui/app/build-configs/directives/pnc-create-build-config-product-version-form/pncCreateBuildConfigProductVersionForm.js:31]: using `const` should be preferred in this case

File: ui/app/common/components/pnc-product-version-combobox/pncProductVersionCombobox.js [hunk 2]
```diff
@@ -91,9 +95,9 @@
       };
 
       function doSearch($viewValue) {
-        var q = rsqlQuery().where('product.name').like('*' + $viewValue + '*').end();
-
-        return ProductVersionResource.query({ q: q }).$promise.then(function (page) {
+        let q = rsqlQuery().where('version').like('*' + $viewValue + '*').end();
+        console.log($ctrl.product+'-'+$ctrl.product.id);
```
Comment (matedo1) [inline ui/app/common/components/pnc-product-version-combobox/pncProductVersionCombobox.js:99]: I guess `console.log` should be removed

File: ui/app/common/pnc-client/resources/ProductVersionResource.js [hunk 3]
```diff
@@ -34,10 +34,6 @@
         id: '@id'
       },
       {
-        query: {
```
Comment (alexcreasy) [inline ui/app/common/pnc-client/resources/ProductVersionResource.js:37]: Why is it necessary to remove this method?

Comment (DnsZhou) [reply ui/app/common/pnc-client/resources/ProductVersionResource.js:37]: @alexcreasy I thought this is for the endpoint: "GET /product-versions" which removed from the backend(as we discussed in NCL-5667"), but let me know if that is not for this endpoint or it should remain there for redundancy reason. Thanks, Alex!

**Comment by DnsZhou** [comment]
retest this please

File: ui/app/common/pnc-client/resources/ProductVersionResource.js (hunk 3 above)
Comment (alexcreasy) [reply ui/app/common/pnc-client/resources/ProductVersionResource.js:37]: Yes, good point, I'd forgotten that! :)

**Comment by cmoulliard** [comment]
When this PR will be merged ?

**Comment by alexcreasy** [comment]
> When this PR will be merged ?

Probably sometime this week. The CI failure will need to be investigated first though. I can't see why there would be much of a delay, although this change is purely for 2.0 so there's no immediate rush to merge it. 

Can I ask what your interest in this PR is? We're simply updating the UI to work correctly with the PNC 2.0 REST API. If you require fixes / improvements to the PNC UI as a priority our sustaing team could look into this for you?

**Comment by janinko** [comment]
retest this please

**Pull request merged**